### PR TITLE
Add tests for relevant optimizer updates in test infra.

### PR DESCRIPTION
**Pull request by umalesTT**
Currently working on bringing up training of BERT, LLaMA and ResNet-18 and planning to add tests for their optimizers within the test infrastructure.
Some basic cases of optimizer updates are added as well, while in all tests gradients are being scaled down by 0.01, given that gradients are generally smaller than numbers generated from random distribution.

### CODE REVIEW

**Review by kmitrovicTT — APPROVED** [review]

File: tests/jax/single_chip/graphs/test_optimizers.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    """Test SGD optimizer update step."""
+    
+    def sgd_update(params: jax.Array, grads: jax.Array) -> jax.Array:
+        learning_rate = 0.01
+        grads = grads * 0.01
+        optimizer = optax.sgd(learning_rate=learning_rate)
+        updates, _ = optimizer.update(grads, optimizer.init(params))
+        return optax.apply_updates(params, updates)
+    
+    run_graph_test_with_random_inputs(sgd_update, [param_shape, param_shape])
```
[the remark this answers is not in the document]
Comment (mrakitaTT) [reply tests/jax/single_chip/graphs/test_optimizers.py:27]: You can create helper function which prepares inputs and gradients tensor, and use it for all these tests, in combination with `run_graph_test` as Stefan suggested, to avoid situation where bug in multiply would affect these tests.

**Review by nsumrakTT — APPROVED** [review]
